operator/cmd/mesh/manifest-apply.go

```diff
@@ -155,6 +155,10 @@ func runApplyCmd(cmd *cobra.Command, rootArgs *rootArgs, maArgs *manifestApplyAr
 func ApplyManifests(setOverlay []string, inFilenames []string, force bool, dryRun bool, verbose bool,
 	kubeConfigPath string, context string, wait bool, waitTimeout time.Duration, l *Logger) error {
 
+	if err := ValidateSetFlags(setOverlay); len(err) != 0 {
```

Comment:
This seems like  the wrong entrypoint, as we want this for `generate` as well I think? I was expecting this to be in the same location as ValidateConfig which has a similar function `deprecatedSettingsMessage`. Its possible those are also at the wrong spot though, but it seems wherever they are they should be co-located?

Reply:
As far as I understood, By looking at folder structure  `operator/cmd/mesh`, I think [apply](https://github.com/istio/istio/blob/b96cec24adbbf2f17b5ec835eef1ec777e81d9f8/operator/cmd/mesh/manifest-apply.go#L155), [generate](https://github.com/istio/istio/blob/4ebdc655e19401b0e616884eca34a586940f0365/operator/cmd/mesh/manifest-generate.go#L92) and [upgrade](https://github.com/istio/istio/blob/4c4cd1c3f6bdc46487886ab29fb63ce5ed0b85ca/operator/cmd/mesh/upgrade.go#L124) all come under the same package name `mesh`. It would be easier to include `ValidateSetFlags` for `generate` and `upgrade` and avoid the import path.

But definitely, we can move it to a better place if it make sense.

Reply:
I am not super familiar with the code here, so I will let others respond the best location, just think wherever it is should be the same place. @ostromart ?